### DISCUSSION TITLE
Update v6.6 - dev.8

### DISCUSSION THREAD

**Discussion by TBog** [TBog/TBLauncher#214] [Pre-Release]
* faster app start (use DB cached apps)
* remove "Favorites"
* fix restore default name for filters and actions
* allow tags as initial result list
* don't record tags in launch history
* show icon pack icon in settings
* fix crash when selecting default icon on tags
* fix tag icon not updating in the tags manager
* use tag icon in settings

<hr /><em>This discussion was created from the release <a href='https://github.com/TBog/TBLauncher/releases/tag/v6.6-dev.8'>Update v6.6 - dev.8</a>.</em>

**Comment by TBog** [comment]
Except for the ripple bug if there are no more issues I'm going to make a stable release out of this.

**Comment by ltguillaume** [comment]
- **Regression** It seems that now it's impossible to show the tags as result list: you can only choose a single tag for a gesture
- **Consistency** You're using both "quick list" and "Quick List", in the same popup menu. Maybe it's just better to rename "quick list" to "dock", since most people know what a dock is in a launcher and the quick list basically is just that? It also has the benefit of not making the popup so wide.
- **Small bug** Choose "Uninstall" and then "Cancel": app is still gone from the list until you reopen or start searching.

**Reply by TBog** [reply]
Unfortunately I can't really quickly find out if you pressed Cancel or not. I'll think of an alternative...

**Reply by ltguillaume** [reply]
This is what I would probably come up with (as some pseudo code):
```
uninstall(position) {
  uninstalling = resultList(position).packageName
  startUninstallIntent
}

onResume {
  if (uninstalling) {
    if (!stillExists(packageName))
      removeFromList(packageName)
    uninstalling = null
  }
}
```

**Reply by TBog** [reply]
As far as I can tell, at least on my device, uninstalling is done in the background and is not a fast process and so the launcher resumes before the app vanishes.
I do have a broadcast receiver from KISS that should handle this but I had some issues in the past with it. Don't remember specifics.
I'll test it out again. Here is the commit 4737a93e58e43790317269fd47208e577a6bb400

**Reply by ltguillaume** [reply]
Aaah yes that makes total sense. Hope the broadcast receiver now works, would indeed be the best solution then.

**Reply by ltguillaume** [reply]
Seems fixed in 6.7

**Comment by ltguillaume** [comment]
Could you explain to me what I should/should not see with the option "Show "Quick List" shortcut badge"? I _guess_ I'm not seeing a difference because I set custom icons?

**Reply by TBog** [reply]
the shortcut badge is a small icon with the original app the shortcut was created from.
That option should show or hide that small icon in the QuickList

**Reply by ltguillaume** [reply]
Aaah ok, so it's "Show shortcut badges _in_ Quick List". Got it!

From the way it's written, I thought it would be a badge in the result list to indicate that the app was in the Quick List, which seemed odd.

**Comment by ltguillaume** [comment]
I just found out that with (one of) the latest dev build(s), the keyboard also shows when you open a tag (or the tags list) when not yet in search mode. I don't think people would want to search by text if they explicitly open a group of apps, because that searches through _all_ apps. I think that before, the keyboard wouldn't open when opening tags from the tags menu, or directly from the dock (when you pin the dock to the widget mode, for example), right?

It may even be a good idea to explicitly close the keyboard if people click on a tag (or the tags list) from the dock, when already in search mode.
